packages/realm/src/errors.ts

```diff
@@ -138,3 +141,45 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to access.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of information about each object that caused the compensating write.
+   */
+  public writeErrors: CompensatingWriteInfo[] = [];
+
+  /** @internal */
+  constructor(error: binding.SyncError) {
+    super(error);
+    for (const write of error.compensatingWritesInfo) {
+      const primaryKey = write.primaryKey;
+      assert.primaryKey(primaryKey);
+      this.writeErrors.push({ objectName: write.objectName, reason: write.reason, primaryKey: primaryKey });
+    }
```

Comment:
```suggestion
    for (const { objectName, primaryKey, reason } of error.compensatingWritesInfo) {
      assert.primaryKey(primaryKey);
      this.writeErrors.push({ objectName, reason, primaryKey });
    }
```

Reply:
I need to remind myself this is javascript...
Fixed it